### ISSUE TITLE
Recent versions don't build Debian packages

### ISSUE THREAD

**Issue by c-hoeger** [openSUSE/obs-build#720]
With the latest OBS installation on Leap 15.3, OBS 2.10.11, Debian package building does no longer work:

```
# rpm -qf /usr/lib/build/init_buildsystem
build-20210830-lp153.372.1.noarch
```

Build log:

```
[    0s] Memory limit set to 5435192KB
[    0s] Using BUILD_ROOT=/srv/obs/tmp/root_1
[    0s] Using BUILD_ARCH=x86_64:i686:i586:i486:i386
[    0s] 
[    0s] 
[    0s] obs-worker-4 started "build open-xchange_7.10.6-0.dsc" at Wed Sep  1 08:17:04 UTC 2021.
[    0s] 
[    0s] Building open-xchange for project 'backend-develop' repository 'DebianBuster' arch 'x86_64' srcmd5 '4b5e26f7d3e9767156d14ab83323b8e4'
[    0s] 
[    0s] processing recipe /srv/obs/tmp/root_1/.build-srcdir/open-xchange_7.10.6-0.dsc ...
[    0s] running changelog2spec --target debian --file /srv/obs/tmp/root_1/.build-srcdir/open-xchange_7.10.6-0.dsc
[    0s] init_buildsystem --configdir /srv/obs/run/worker/1/build/configs --cachedir /var/cache/build --clean --rpmlist /srv/obs/tmp/root_1/.build.rpmlist /srv/obs/tmp/root_1/.build-srcdir/open-xchange_7.10.6-0.dsc build ...
[    1s] cycle: libc6 -> libgcc1
[    1s]   breaking dependency libgcc1 -> libc6
[    1s] [1/71] preinstalling e2fslibs...
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] [2/71] preinstalling gcc-8-base...
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] [3/71] preinstalling libsemanage-common...
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] [4/71] preinstalling lsb-base...
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] [5/71] preinstalling ncurses-base...
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] [6/71] preinstalling sensible-utils...
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] [7/71] preinstalling libgcc1...
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] usage: unpackarchive --cpio|--tar <root>
[    1s] [8/71] preinstalling libc6...
[    1s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [9/71] preinstalling bsdutils...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [10/71] preinstalling debianutils...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [11/71] preinstalling diffutils...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [12/71] preinstalling hostname...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [13/71] preinstalling insserv...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [14/71] preinstalling libattr1...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [15/71] preinstalling libbz2-1.0...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [16/71] preinstalling libc-bin...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [17/71] preinstalling libcom-err2...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [18/71] preinstalling libdb5.3...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [19/71] preinstalling libdebconfclient0...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [20/71] preinstalling liblzma5...
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] usage: unpackarchive --cpio|--tar <root>
[    2s] [21/71] preinstalling libncurses5...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [22/71] preinstalling libpcre3...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [23/71] preinstalling libsepol1...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [24/71] preinstalling libsigsegv2...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [25/71] preinstalling libslang2...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [26/71] preinstalling libsmartcols1...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [27/71] preinstalling libtinfo6...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [28/71] preinstalling libustr-1.0-1...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [29/71] preinstalling libuuid1...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [30/71] preinstalling multiarch-support...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [31/71] preinstalling zlib1g...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [32/71] preinstalling base-passwd...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [33/71] preinstalling libacl1...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [34/71] preinstalling libblkid1...
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] usage: unpackarchive --cpio|--tar <root>
[    3s] [35/71] preinstalling libselinux1...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [36/71] preinstalling libss2...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [37/71] preinstalling ncurses-bin...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [38/71] preinstalling xz-utils...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [39/71] preinstalling libperl5.28...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [40/71] preinstalling findutils...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [41/71] preinstalling libmount1...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [42/71] preinstalling sed...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [43/71] preinstalling tar...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [44/71] preinstalling coreutils...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [45/71] preinstalling e2fsprogs...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [46/71] preinstalling libsemanage1...
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] usage: unpackarchive --cpio|--tar <root>
[    4s] [47/71] preinstalling dpkg...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [48/71] preinstalling readline-common...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [49/71] preinstalling gzip...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [50/71] preinstalling perl-base...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [51/71] preinstalling grep...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [52/71] preinstalling debconf...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [53/71] preinstalling init-system-helpers...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [54/71] preinstalling libreadline7...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [55/71] preinstalling tzdata...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [56/71] preinstalling libpam0g...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [57/71] preinstalling gawk...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [58/71] preinstalling dash...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [59/71] preinstalling libpam-modules-bin...
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] usage: unpackarchive --cpio|--tar <root>
[    5s] [60/71] preinstalling libpam-modules...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [61/71] preinstalling libpam-runtime...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [62/71] preinstalling passwd...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [63/71] preinstalling user-setup...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [64/71] preinstalling login...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [65/71] preinstalling base-files...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [66/71] preinstalling util-linux...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [67/71] preinstalling sysvinit-utils...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [68/71] preinstalling bash...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [69/71] preinstalling mount...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [70/71] preinstalling sysv-rc...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] [71/71] preinstalling initscripts...
[    6s] usage: unpackarchive --cpio|--tar <root>
[    6s] usage: unpackarchive --cpio|--tar <root>
[    7s] reordering...cycle: libc6 -> libgcc1
[    7s]   breaking dependency libgcc1 -> libc6
[    7s] cycle: debhelper -> dh-autoreconf
[    7s]   breaking dependency debhelper -> dh-autoreconf
[    7s] cycle: debhelper -> dh-strip-nondeterminism
[    7s]   breaking dependency debhelper -> dh-strip-nondeterminism
[    7s] done
[    7s] querying package ids...
[    9s] [1/192] installing autotools-dev-20180224.1
[    9s] chroot: failed to run command 'dpkg': No such file or directory
[    9s] exit ...
[    9s] 
[    9s] obs-worker-4 failed "build open-xchange_7.10.6-0.dsc" at Wed Sep  1 08:17:13 UTC 2021.
[    9s] 
```

**Comment by c-hoeger** [comment]
Workaround

```diff
buildservice:/usr/lib/obs/server # diff -u /usr/lib/build/init_buildsystem.orig /usr/lib/build/init_buildsystem
--- /usr/lib/build/init_buildsystem.orig	2021-09-01 15:07:01.250633215 +0200
+++ /usr/lib/build/init_buildsystem	2021-09-01 14:51:44.885306100 +0200
@@ -282,9 +282,9 @@
     else
 #	unsafe_preinstall_check
 #	CPIO="cpio --extract --unconditional --preserve-modification-time --make-directories --no-absolute-filenames --quiet"
-#	TAR="tar --exclude .build --exclude .init_b_cache -x"
+	TAR="tar --exclude .build --exclude .init_b_cache -x"
 	CPIO="$BUILD_DIR/unpackarchive --cpio ."
-	TAR="$BUILD_DIR/unpackarchive --tar ."
+	#TAR="$BUILD_DIR/unpackarchive --tar ."
     fi
     pkg_preinstall
     preinstall_integrate
```

**Comment by mlschroe** [comment]
Whoa, do you really run an OBS worker without bsdtar installed?

**Comment by mlschroe** [comment]
i.e. only chroot builds?

**Comment by c-hoeger** [comment]
> Whoa, do you really run an OBS worker without bsdtar installed?

Hm, well, yes, never installed that before. Wasn't even aware that it exists on Linux...
But yes, once installed (package libarchive-tools), it works again.
Much better workaround ;)

**Comment by mlschroe** [comment]
bsdtar supports the --chroot option, which makes package unpacking much less error prone.

Anyway, unpacking without bsdtar has been fixed with commit 7512dbf6ad8d6642a0e62bd31266a05b868add4c. Sorry for breaking it in the first place, I should have tested the commit much better.
